index.d.ts

```diff
@@ -115,16 +123,22 @@ declare namespace Cmd {
     args?: any[]
   ): MapCmd<A>;
 
-  export function run<A extends Action, B extends Action>(
-    f: Function,
+  export function run<
+    TSuccessAction extends Action,
+    TFailAction extends Action,
+    TFunction extends (...args: any[]) => any | Promise<any>
+  >(
+    f: TFunction,
     options?: {
       args?: any[];
```

Comment:
Thanks for the improvement to the types. I've been waiting for this for a long time. I'd suggest a small addition to the types:

The introduction of the `TFunction` type argument could also enable better type safety of the function arguments. So the type for the `args` option would be better defined as:

```typescript
  args?: Parameters<TFunction>;
```